tests/scripts/multus/rook-ceph-public-cluster-nads.yaml

```diff
@@ -15,7 +15,7 @@ spec:
       "mode": "bridge",
       "ipam": {
         "type": "whereabouts",
-        "range": "192.168.20.0/24",
+        "ipRanges": [{"range": "192.168.20.0/29}, {"range": "192.168.25.0/24"}],
```

Comment:
The test seems to be failing with this NetAttDef config, am I interpreting that right?

I would expect this to succeed. There are plenty of IPs in the combined range. It's possible there is a whereabouts issue, but I think that's probably not very likely.

Let's try to dig into this together to see if we can figure out what's happening.

Reply:
your interpetation is right...let's debug this in the call?

Reply:
I confirmed with a Whereabouts dev that the behavior we want is not the behavior Whereabouts gives us. This isn't a bug in whereabouts but an issue of our understanding.

With the `ipRanges` config, Whereabouts will give multiple IPs **at the same time**. My assumption was that this was a piecewise range where it would use the second range after the first ran out, but that isn't the case. Whereabouts doesn't support piecewise ranges currently, nor does that seem to be a request that Whereabouts has gotten before.

If users require piecewise ranges, we should recommend them use DHCP instead. Whenever users use DHCP, it is best if they use `addressRanges`.

For our testing in CI, it is okay for us to use a single whereabouts `range` and then over-specify `addressRanges`. As long as we verify that the `addressRanges` specified in CephCluster.spec.network appear in the `ceph config dump`, we can be quite sure the feature is working as intended.